src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add actor " + actor + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
```

Comment:
As per the checklist, you don't need to set an ID for your entity explicitly as it will be done by Hibernate. Ensure that the Actor entity is properly annotated to let Hibernate handle ID generation.